UnityProject/Assets/Scripts/Mobs/BrainAI/States/SimpleBot/SimpleBotTaskAi.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using AddressableReferences;
+using Mirror;
+using NaughtyAttributes;
+using Player.Language;
+using UnityEngine;
+using UnityEngine.Serialization;
+
+namespace Mobs.BrainAI.States.SimpleBot
+{
+	[System.Serializable]
+	public struct BotDialogue
+	{
+		public AddressableAudioSource audioSource;
+		[FormerlySerializedAs("Transcription")] public string transcription;
+	}
+
+	public class SimpleBotTaskAi : BrainMobState
+	{
+
+		[SerializeField] protected FindSimpleTaskAi findSimpleTaskAi = null;
+		[SerializeField] protected AddressableAudioSource taskPerformSound;
+		[SerializeField] protected AddressableAudioSource emaggedPerformSound;
+
+		[SerializeField] protected float taskPerformDuration = 2f;
+
+		//How far should this bot search for a target? Consider using lower values for less performant checks
+		protected int searchRadius = 5;
+
+		protected Matrix targetMatrix;
+		protected Vector3Int targetCell;
+
+		[field: SyncVar] public bool IsEmagged { get; private set; } = false;
+		protected Coroutine taskPerformCoroutine = null;
+
+		public delegate void SpriteChangeEvent(bool isEmagged, bool isPerformingTask);
+		public SpriteChangeEvent OnSpriteChange = null;
+
+		[SerializeField] protected LanguageSO botLanguage = null;
+		[SerializeField] protected List<BotDialogue> stateExitDialogue = new List<BotDialogue>();
+
+
+		[Button()]
+		public void ToggleEmagState()
+		{
+			SetEmagState(!IsEmagged);
+		}
+
+		public void SetEmagState(bool state)
```

Comment:
I would say maybe have a generic body partemagble but since it's only used in one place it's fine to have a Singular script for it

Reply:
What you mean? This emag state is called from the ICanBeEmaggedMob component

Reply:
oh ok, It's fine to use a non-interface version of this since, the body parts act as interfface bit bt that should be all ggood